### PR TITLE
Suggested rollout plan

### PR DESCRIPTION
**Pull request by trieloff**
As a starting point for discussions, here is the rollout plan I'd suggest.

### CODE REVIEW

**Comment by trieloff** [comment]
See also #134 #128

File: ROLLOUT.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+
+**Note:** the goal of this staged rollout is to provide a stable experience for the majority of visitors, with the ability to expand the new homepage experience to greater visitor groups once it has been validated.
+
+Because the new experience is only for the home page, redirects don't have to be migrated yet, as all potential redirects affect non-homepage content only.
+
+#### Prerequisites
+
+- [ ] a `home-geo-ch` strain pointing to the new site
+- [ ] a finished home page implementation
+- [x] Helix team on standby in `#helix-escalations` Slack channel
```
Comment (filmaj) [inline ROLLOUT.md]: Can someone make this channel shared org-wide? @trieloff

File: ROLLOUT.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+- [ ] a basic functional test for the new home page, running in development mode
```
Comment (filmaj) [inline ROLLOUT.md]: I _think_ we have something like that under `tests/` already, ensuring homepage content matches the markdown.

File: ROLLOUT.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+As a quick fix for most problems, we will establish Fallback and Rollback procedures that can alleviate errors until a more permanent fix has been provided.
+
+### Fallback
+
+Turn off the failing strain, so that the `default` (proxy to existing AEM) gets applied, showing the "old" version of the site.
+
+### Rollback
+
+Perform a `hlx publish` using an older `helix-config.yaml` that has been committed as part of an older, successful rollout.
```
Comment (filmaj) [inline ROLLOUT.md]: Can we expand on how this would work in practice with version control? Would CircleCI, then, be able to push changes to master to enable such rollbacks?

File: ROLLOUT.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+## Monitoring and Escalation
+
+### Monitoring
+
+The existing monitoring procedures that are used for the AEM-based site stay in place. As the URLs don't change and monitoring is technology insensitive, this enables us to have the same level of quality as before.
+
+In the future, more advanced monitoring techniques can be applied.
+
+### Escalation
```
Comment (filmaj) [inline ROLLOUT.md]: I'd ask that as part of the escalation procedure we formalize updating the or creating an operational runbook for the particular symptom / incident. A big inspiration to me is how gitlab does it with their open runbooks: https://gitlab.com/gitlab-com/runbooks

Comment (rofe) [reply ROLLOUT.md]: +1

File: ROLLOUT.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+
+The Helix team is providing and monitoring the `#helix-escalations` Slack channel, which will also perform as a war room in case of escalation.
+
+In the beginning, this should even be done for escalations that only affect the existing infrastructure, so that the Helix team can develop procedures that allow a quick assertion if an issue is caused by:
+
+- strains running on Project Helix
+- or strains running as proxy strains, using the existing stack
+- or issues in the strain resolution, or proxying itself
+
+Only in the second case the Helix team will be unable to provide assistance.
```
Comment (rofe) [inline ROLLOUT.md]: I would rephrase this to "Only in the second case the Helix team will only be able to provide limited assistance."

File: ROLLOUT.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+The existing monitoring procedures that are used for the AEM-based site stay in place. As the URLs don't change and monitoring is technology insensitive, this enables us to have the same level of quality as before.
+
+In the future, more advanced monitoring techniques can be applied.
+
+### Escalation
+
+The existing escalation paths stay in place, with one modification: **For every escalation, the Helix team will be involved**.
+
+The Helix team is providing and monitoring the `#helix-escalations` Slack channel, which will also perform as a war room in case of escalation.
+
```
Comment (rofe) [inline ROLLOUT.md]: For the likely event that Slack notifications go unnoticed for more than 5 minutes (I'd say it happened to all of us :P) I would add that the Helix team will also provide access to a roster of people on-call, so one can resort to other means to grab their attention.

File: ROLLOUT.md (hunk 1 above)
Comment (filmaj) [reply ROLLOUT.md]: NM I think that's done.

File: ROLLOUT.md (hunk 3 above)
Comment (trieloff) [reply ROLLOUT.md]: The idea would be that after running `hlx publish`, you'd `git ci -m "[ci skip]"` back, so that we have the state of the `helix-config.yaml` captured.

File: ROLLOUT.md (hunk 4 above)
Comment (trieloff) [reply ROLLOUT.md]: This one is great, too: https://response.pagerduty.com

File: ROLLOUT.md (hunk 3 above)
Comment (filmaj) [reply ROLLOUT.md]: So a rollback would mean `git revert`ing any commits that landed since the `[ci skip]` commit freezing the state of the `helix-config.yaml` ?

**Comment by trieloff** [comment]
Suggestion from @davidnuescheler: we might want to start with a single tech doc page tree instead of the home page, as the stakeholder demands will be smaller.

**Comment by filmaj** [comment]
Need to pull in @simonwex regarding that suggestion from David, FYI Simon

File: ROLLOUT.md [hunk 7]
```diff
@@ -0,0 +1,200 @@
...
+- [ ] Fastly configuration receives traffic for `www.adobe.io` domain
+- [ ] CircleCI is set up to run `hlx publish` on `master` (see
+    [#152](https://github.com/adobe/developer.adobe.com/pull/152) for
+    work-in-progress proposal.
+- [ ] GitHub `master` branch is protected from accidental merges
+
+### Phase 2: New home page for some
+
+*TODO*: Should the target first page for the rollout be the home page? @davidnuescheler
+has opined that this should be a subtree to begin with.
```
Comment (filmaj) [inline ROLLOUT.md]: @simonwex what do you think about this suggestion from @trieloff ?

File: ROLLOUT.md [hunk 8]
```diff
@@ -0,0 +1,200 @@
...
+
+The requirement for uptime is 99.99% which allows for 52 minutes and 36 seconds
+of downtime per year.
+
+A "publishing failure" should be considered as downtime. Time-to-live should be
+taken into account and clearly documented for content repository updates, but
+this TTL should be treated as a service-level agreement between us (the site) and
+our customers (content authors).
+
+*TODO*: can we define a 'publishing failure' more completely?
```
Comment (filmaj) [inline ROLLOUT.md]: @simonwex may I ask for your help in tackling this TODO item?

File: ROLLOUT.md [hunk 9]
```diff
@@ -0,0 +1,200 @@
...
+
+### Implementation
+
+The existing monitoring procedures that are used for the AEM-based site stay in
+place. As the URLs don't change and monitoring is technology insensitive, this
+enables us to have the same level of quality as before.
+
+*TODO*: What are the existing monitoring procedures? @kgera?
```
Comment (filmaj) [inline ROLLOUT.md]: @kgera your help in documenting or listing the existing monitoring in place for the AEM would be greatly appreciated here.

File: ROLLOUT.md [hunk 10]
```diff
@@ -0,0 +1,200 @@
...
+
+*TODO*: should the helix-powered site / team / this project aim to implement its
+own monitoring?
+
+## Escalation
+
+The existing escalation paths stay in place, with one modification: **For every
+escalation, the Helix team will be involved**.
+
+*TODO*: What are the existing escalations? @kgera?
```
Comment (filmaj) [inline ROLLOUT.md]: @kgera your help in documenting the existing escalation procedures would be greatly appreciated here.

File: ROLLOUT.md [hunk 11]
```diff
@@ -0,0 +1,200 @@
...
+
+The Helix team is providing and monitoring the `#helix-escalations` Slack channel,
+which will also perform as a war room in case of escalations. Every message in
```
Comment (filmaj) [inline ROLLOUT.md]: @trieloff since the `#helix-escalations` channel creates a notification for _every_ message, I suggest we reconsider using this same channel for the war room.

File: ROLLOUT.md (hunk 6 above)
Comment (kgera) [reply ROLLOUT.md]: Also, a weekend support on-call roster will be good to have for atleast initial few weeks.

File: ROLLOUT.md (hunk 10 above)
Comment (kgera) [reply ROLLOUT.md]: Currently, we get notifications and alerts from AMS on P2 priority issues and they are addressed by the website team engaging AEM support team as needed. 
For Any P1 issue, the AMS actions on the resolution engaging support teams and also NOC notifies the website team on the details of P1 issue and a war room is opened as needed.

File: ROLLOUT.md (hunk 7 above)
Comment (kgera) [reply ROLLOUT.md]: +1 here. My two cents, I think probably going live with any API documentation page or support/open source page will be a safer bet. Also, analytics and SEO is something we have to keep in mind, if we can make the support/open source page live as it has no dependency on any external stakeholder since website team has the ownership for those 2 pages.

File: ROLLOUT.md (hunk 11 above)
Comment (kgera) [reply ROLLOUT.md]: Existing escalations are generally initiated to NOC teams. However, in case any downtime happens a CSO is opened and the CSO team opens a war room with all involved stakeholders they have listed. We will need to work on those details.

File: ROLLOUT.md (hunk 10 above)
Comment (filmaj) [reply ROLLOUT.md]: No need for a different document, I'll incorporate your response into this document directly.

File: ROLLOUT.md (hunk 11 above)
Comment (filmaj) [reply ROLLOUT.md]: Yes it would be good to get those details and re-incorporate into this document.

Comment (trieloff) [reply ROLLOUT.md]: Yes, we are thinking about using threads instead. https://github.com/adobe/project-helix/pull/382/files#r261074126

File: ROLLOUT.md (hunk 6 above)
Comment (filmaj) [reply ROLLOUT.md]: In terms of helix support / on-call, that is an ongoing conversation in  https://github.com/adobe/project-helix/pull/382, which I've noted as a todo in the document to fold results of that back into this proposal.

**Comment by filmaj** [comment]
Making good progress here but still a few unresolved conversations - @simonwex @kgera and @trieloff's  involvement in those  would be greatly appreciated.
